internal/serve/httphandler/assets_handler.go

```diff
@@ -131,12 +137,18 @@ func (c AssetsHandler) DeleteAsset(w http.ResponseWriter, r *http.Request) {
 			return nil, fmt.Errorf("error performing soft delete on asset id %s: %w", assetID, deleteErr)
 		}
 
-		trustlineErr := c.handleUpdateAssetTrustlineForDistributionAccount(ctx, nil, &txnbuild.CreditAsset{
-			Code:   deletedAsset.Code,
-			Issuer: deletedAsset.Issuer,
-		})
-		if trustlineErr != nil {
-			return nil, fmt.Errorf("error removing trustline: %w", trustlineErr)
+		distributionAccount, getDistAccFromCtxErr := c.DistributionAccountResolver.DistributionAccountFromContext(ctx)
+		if getDistAccFromCtxErr != nil {
+			return nil, fmt.Errorf("resolving distribution account from context: %w", getDistAccFromCtxErr)
+		}
+
+		if distributionAccount.IsStellar() {
+			if trustlineErr := c.handleUpdateAssetTrustlineForDistributionAccount(ctx, nil, &txnbuild.CreditAsset{
+				Code:   deletedAsset.Code,
+				Issuer: deletedAsset.Issuer,
+			}, distributionAccount); trustlineErr != nil {
+				return nil, fmt.Errorf("error removing trustline: %w", trustlineErr)
+			}
```

Comment:
spoke with @marwen-abid and he recommends making the check inside `handleUpdateAssetTrustlineForDistributionAccount`